retrofit-converters/jaxb/src/main/java/retrofit2/converter/jaxb/JaxbConverterFactory.java

```diff
@@ -28,29 +30,94 @@
+  public static JaxbConverterFactory create(JAXBContext context, ConvertingMediaType convertingMediaType) {
+    if (context == null) throw new NullPointerException("context == null");
+    if (convertingMediaType == null) throw new NullPointerException("convertingMediaType == null");
+    return new JaxbConverterFactory(context, convertingMediaType,null, null);
+  }
+
+  /** Create an instance using a default {@link JAXBContext}
+   * instance for conversion with custom (un)marshaller properties. */
+  public static JaxbConverterFactory create(Map<String, Object> marshalProps, Map<String, Object> unmarshalProps) {
+    return new JaxbConverterFactory(null, null, marshalProps, unmarshalProps);
+  }
+
+  /** Create an instance using a default {@link JAXBContext}
+   * instance for conversion with custom (un)marshaller properties. */
+  public static JaxbConverterFactory create(
+      ConvertingMediaType convertingMediaType,
+      Map<String, Object> marshalProps,
+      Map<String, Object> unmarshalProps) {
+    if (convertingMediaType == null) throw new NullPointerException("convertingMediaType == null");
+    return new JaxbConverterFactory(null, convertingMediaType, marshalProps, unmarshalProps);
+  }
+
+  /** Create an instance using {@code context} for conversion with custom (un)marshaller properties. */
+  @SuppressWarnings("ConstantConditions")
+  public static JaxbConverterFactory create(
+      JAXBContext context,
+      Map<String, Object> marshalProps,
+      Map<String, Object> unmarshalProps) {
+    if (context == null) throw new NullPointerException("context == null");
+    return new JaxbConverterFactory(context, null, marshalProps, unmarshalProps);
+  }
+
+  /** Create an instance using {@code context} for conversion with custom (un)marshaller properties. */
+  @SuppressWarnings("ConstantConditions")
+  public static JaxbConverterFactory create(
+          JAXBContext context,
+          ConvertingMediaType convertingMediaType,
+          Map<String, Object> marshalProps,
+          Map<String, Object> unmarshalProps) {
+    if (context == null) throw new NullPointerException("context == null");
+    if (convertingMediaType == null) throw new NullPointerException("convertingMediaType == null");
+    return new JaxbConverterFactory(context, convertingMediaType, marshalProps, unmarshalProps);
   }
 
   /** If null, a new JAXB context will be created for each type to be converted. */
   private final @Nullable JAXBContext context;
+  /** If null, will be XML by default */
+  private final ConvertingMediaType convertingMediaType;
+  private final Map<String, Object> marshalProps;
+  private final Map<String, Object> unmarshalProps;
 
-  private JaxbConverterFactory(@Nullable JAXBContext context) {
+  private JaxbConverterFactory(
+      JAXBContext context,
+      @Nullable ConvertingMediaType convertingMediaType,
+      @Nullable Map<String, Object> marshalProps,
+      @Nullable Map<String, Object> unmarshalProps) {
```

Comment:
I would prefer we move default media type and the `emptyMap()`s to the call sites and keep all of these argument non-null.